librz/analysis/var.c

```diff
@@ -722,14 +722,42 @@ static bool stack_offset_is_arg(RzAnalysisFunction *fcn, st64 stack_off) {
 	return stack_off >= shadow;
 }
 
+static bool reg_is_fcn_arg(RzAnalysisFunction *fcn, const char *regname) {
+	if (RZ_STR_ISEMPTY(fcn->cc)) {
+		return false;
+	}
+	int n = rz_analysis_cc_max_arg(fcn->analysis, fcn->cc);
+	for (int i = 0; i < n; ++i) {
+		const char *x = rz_analysis_cc_arg(fcn->analysis, fcn->cc, i);
+		if (strcmp(x, regname) == 0)
+			return true;
+	}
+	return false;
+}
+
 RZ_API bool rz_analysis_var_is_arg(RzAnalysisVar *var) {
```

Comment:
Could you please add Doxygen here while you are on it?

Reply:
also RZ_NONNULL